src/test/operators/aggregate_test.cpp

```diff
@@ -177,6 +182,38 @@ TYPED_TEST(OperatorsAggregateTest, CannotStandardDeviationSampleStringColumns) {
   EXPECT_THROW(aggregate->execute(), std::logic_error);
 }
 
+// The ANY aggregation is a special case which is used to obtain "any value" of a group of which we know that each
+// value in this group is the same (for most cases, the group will have a size of one). This can be the case, when
+// the aggragated column is functionally dependent on the group-by columns.
+TYPED_TEST(OperatorsAggregateTest, FailAnyOnNonDependentColumn) {
+  auto filtered = std::make_shared<TableScan>(
+      this->_table_wrapper_2_2, equals_(this->get_column_expression(this->_table_wrapper_2_2, ColumnID{0}), 123));
+  filtered->execute();
+
+  {
+    auto aggregate = std::make_shared<TypeParam>(
+        filtered, std::vector<AggregateColumnDefinition>{{ColumnID{2}, AggregateFunction::Any}},
+        std::vector<ColumnID>{ColumnID{0}, ColumnID{1}});
+    aggregate->execute();
+
+    EXPECT_EQ(aggregate->get_output()->template get_value<int>(ColumnID{2}, 0u), 20);
```

Comment:
I'd argue that ANY on non-equal values is Hyrise-UB, so `20` is not a "right" result that we should test for.

Reply:
Not sure what you mean. The test first filters the table for the first column being `123`. For all aggregation values of the remaining tuples, the value is 20.

Reply:
`column_2` is `ANY(input.column_2)`, isn't it?

Reply:
Yes.

Reply:
```
=== Columns
|       a|       b|  ANY(c)|
|     int|   float|     int|
|not null|not null|    null|
=== Chunk 0 ===
|<ValueS>|<ValueS>|<ValueS>|
|     123|   458.7|      20|
```

Reply:
Split the tests.